src/components/DemoConfigDialog/DemoConfigDialog.tsx

```diff
@@ -1,80 +1,178 @@
-import React, { MouseEventHandler, useState } from 'react';
+import React, { ChangeEventHandler, MouseEventHandler, useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
+import { useNavigate } from 'react-router';
+import type { UseQueryResult } from 'react-query';
 
 import styles from './DemoConfigDialog.module.scss';
 
 import ErrorPage from '#components/ErrorPage/ErrorPage';
 import TextField from '#components/TextField/TextField';
 import Button from '#components/Button/Button';
-import { addConfigQueryParam, clearStoredConfig, getConfigLocation } from '#src/utils/configOverride';
+import { getConfigNavigateCallback } from '#src/utils/configOverride';
 import Link from '#components/Link/Link';
 import ConfirmationDialog from '#components/ConfirmationDialog/ConfirmationDialog';
+import LoadingOverlay from '#components/LoadingOverlay/LoadingOverlay';
+import type { Config } from '#types/Config';
+import DevStackTrace from '#components/DevStackTrace/DevStackTrace';
 
-const fallbackConfig = import.meta.env.APP_DEMO_DEFAULT_CONFIG_ID;
+const fallbackConfig = import.meta.env.APP_DEMO_FALLBACK_CONFIG_ID;
```

Comment:
I agree, it kind of makes sense to have all the config sources in the same place, but the reason I didn't put it into the settings is that it's a thing that no customers should probably need to touch. Having it in the settings makes it more visible outside of the demo component.  What do you think?

Reply:
We also expose .env files to GitHub, so I think it could be confusing. I would suggest to store it in `templates/webapp.demo.ini` as a default config. The component is named 'DemoConfigDialog' and it seems logical to me that it uses a demo config as a fallback source.